src/base/validator.ts

```diff
@@ -121,3 +122,22 @@ export async function throwErrorAfterUpdateComplete(
   await _this.updateComplete;
   throw new Error(message);
 }
+
+export function validateColumns(columns: Column[]) {
+  if (!Array.isArray(columns)) return false;
+  if (columns[0] === null) return false;
+  return true;
+}
+
+export function validateData(data: object[]) {
+  if (!Array.isArray(data)) return false;
+  if (data[0] === null) return false;
+  return true;
+}
+
+export function validateRowsPerPage(numRows: number) {
+  if (numRows < 0 || numRows === 0 || typeof numRows !== "number") {
```

Comment:
if (numRows < 1 || typeof numRows !== "number") {

Reply:
I think it should rather be this:
if (numRows < 0.5 || typeof numRows !== "number") return false;

Because the ReadOnlyTable allows a number like 0.5 to be rounded to 1.
What do you think?

@nmanhit